### PR TITLE
Updating tf.experimental.numpy.vander for N=0

### LINKED ISSUE #60827: wrong output shape with tf.experimental.numpy.vander when parameter N=0
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

tf 2.12.0

### Custom Code

No

### OS Platform and Distribution

Windows 11

### Mobile device

_No response_

### Python version

3.11

### Bazel version

_No response_

### GCC/Compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current Behaviour?

A bug happened!  when I used tf.experimental.numpy.vander() I get wrong values when parameter N =0  it produce empty array but with different shape that need to be 

### Standalone code to reproduce the issue

```shell
When I run
tf.experimental.numpy.vander(
    [-1.,-1.], N=0, increasing=False
)
I get the output as the following:
-> <tf.Tensor: shape=(2, 2), dtype=float64, numpy=
array([[-1.,  1.],
       [-1.,  1.]])>

But when I run 
np.vander(np.array([-1.,-1.]),N=0)
I got the following: 
-> array([], shape=(2, 0), dtype=float64)
```

### Relevant log output

```shell
tensorflow output 
<tf.Tensor: shape=(2, 2), dtype=float64, numpy=
array([[-1.,  1.],
       [-1.,  1.]])>
numpy output 
array([], shape=(2, 0), dtype=float64)
```
</details>

### PR DESCRIPTION
**Pull request by SuryanarayanaY**
At present the API `tf.experimental.numpy.vander` behaves differently compared to its Numpy variant `numpy.vander` when the argument value of `N=0` .

I think the behaviour of Numpy variant is correct.Current TF implementation broadcasts N to `N=shape(x)[0]` when either N=0 or N=None which is not correct. When N=0 there should not be any broadcasting which is equivalent to `numpy.vander` behaviour.Hence proposing the changes in code to get numpy behaviour.

Attached the gist for demo in same with and without the code changes and with the proposed change in code numpy behaviour can be achievable.

Fixes #60827

### CODE REVIEW

File: tensorflow/python/ops/numpy_ops/np_array_ops.py [hunk 1]
```diff
@@ -1367,8 +1367,9 @@ def vander(x, N=None, increasing=False):  # pylint: disable=missing-docstring,in
   x = asarray(x)
 
   x_shape = array_ops.shape(x)
-  N = N or x_shape[0]
-
+  if N != 0:
```
Comment (cantonios) [inline tensorflow/python/ops/numpy_ops/np_array_ops.py:1370]: For better clarity you should probably just use the simple if statement:
```
if N is None:
  N = x_shape[0]
```

**Review by cantonios — CHANGES_REQUESTED** [review]
Can you add a test to verify the new behavior?

**Comment by gbaned** [comment]
Hi @SuryanarayanaY Can you please check @cantonios's [comments](https://github.com/tensorflow/tensorflow/pull/61079#pullrequestreview-1498899933) ? Thank you!

**Comment by SuryanarayanaY** [comment]
Changes done as requested. Thanks!

**Review by cantonios — CHANGES_REQUESTED** [review]
Still awaiting a test for the new behavior

**Comment by gbaned** [comment]
Hi @SuryanarayanaY Any update on this PR? Please. Thank you!

**Review by cantonios — APPROVED** [review]

**Pull request merged**